addons/web/static/src/core/dialog/dialog.xml

```diff
@@ -30,7 +30,7 @@
                                 </button>
                             </t>
                         </footer>
-                        <main class="modal-body" t-att-class="{'p-0': !props.withBodyPadding}">
+                        <main class="modal-body" t-att-class="props.withBodyPadding ? 'hasBodyPadding' : 'p-0'">
```

Comment:
I'm not sure about our className conventions, but I don't think this one follows them.